### ISSUE TITLE
dialling numbers shorter than 7 digits?

### ISSUE THREAD

**Issue by jaydrive2000** [SigmazGFX/RotaryLockSD3#1]
So i'm going to be attempting a build at this project, and one thing i thought may be useful is that users could dial a number shorter than 7 digits. I've not had opportunity to try this due to waiting on parts, but you could say as a starting clue "If you need any help, dial 100 for information" as an example. There is no reference in the documentation to say if a 3 digit number is supported. Maybe you could clarify? Thankyou.

**Comment by SigmazGFX** [comment]
Hey Jonathon,
First let me start off by saying "Thank you!",  and I'm honored that you think this project is worthy or reproduction.
As to your question, Yes and No.. The current code is written to support 7 digit numbers only but that limitation was because the original purpose of the build called for 7 digit codes. However there are two approaches that can be taken to customize it to work with any number of digits both with pros and cons.

**Create a variable to define the number of digits expected.**
Pro:   This would be something that would be defined on the SD card in the config file and could be changed at any time based on the project needs. Simple implementation.
 
Con:   Still limits the number of digits dialed before an action to the defined number of digits. 


**Change the code to not look for a finite number of digits before acting upon the entry but to employ a timeout delay after the last digit is dialed.**

Pro:   This would allow any variable number of digits to be used up to the max size of the array

Con(sort of):   User actions may be misdirected if they are too slow to dial the next digit in the sequence or string (which could be countered by making sure none of the expected digit strings are similar, (i.e. 123, 12345, 1234567...) instead by making sure to use a unique construct with each target number.
Another disadvantage to this approach is that while the user is waiting for the last digit timeout to lapse the user may feel that the entry wasn't accepted and they may hang up to try again. 
Mainly these (non)issues all revolve around timing and user interaction.

I have been planning on inserting some faint Crossbar or ESS signalling sounds after the last digit has been dialed to truly emulate the vintage dialing experience and make it seem as authentic as possible.

Although the project page doesn't mention it, (Because I suck at documentation) The phone does have sidetone (Side tone is when you can hear your own speech in the earpiece) which lends to the illusion that you're working on an authentic vintage telephone connected to a POTS line.
The only other enhancement which I haven't added yet is for the phone to pulse in the earpiece while dialing. 

But again, I am thrilled that you are interested in this project and if I can help in any way feel free to ask.

**Comment by jaydrive2000** [comment]
Hi again -

Just as i got the notification of your response, my arduino Uno's arrived, so i'm guessing it was meant to be!

I love the idea of this project because of the re-use of old tech and giving it new life, and the customisation options, so looking forward to it. My handset is ordered and on the way however!

This now brings me onto the Arduino itself. You've used an arduino Uno for your build. I have some Arduino Nano's and Arduino Pro Mini's which are essential the Uno but in a different form - i tried burning the hex file to a Nano, but as expected the HEX file supplied is expecting the chip signature of the Uno's IC, and therefore causing a failure to upload. Do you think you could push out some HEX files for the Nano and Pro Mini as they are far smaller and help save on space in the handset device?

**Comment by SigmazGFX** [comment]
Ahh yes.. 
The Nano, although the processor is the same the board does break out 2 additional analog pins (which could be very handy!)
I am at work right now and don't have my project files with me..... Hmm.... wait.. maybe on one of my USB fobs.. 
Let me see if I can get it together for you.. if not right now, later tonight If I can remember..

**Comment by SigmazGFX** [comment]
One question, IIRC the Nano used the CH340 driver for its USB interface chip. 
are you able to load sketches to it from the Arduino IDE?
If not, you may want to get the communications to the Nano part working first..
But yes, a Nano is a great substitute for this project, especially if you want to install a battery pack inside the phone and/or a SLIC to make the factory bells ring.. 
Everything could fit nicely in the space of the original phone network block.
(now you've got me thinking about a RotaryLockSD-4 again.. lol)

**Comment by jaydrive2000** [comment]
Yes, I'm able to load sketches to nano's with the CH340 driver - but no rush on pushing out updates, but i'm glad you've found some inspiration for another major revision to go out =)

**Comment by SigmazGFX** [comment]
Hey Jay! I'm so sorry that I haven't been able to update this yet.
I am over here wondering how you've made out with this project
Reach out to me via email and we can hopefully get your requests fulfilled

**Comment by jaydrive2000** [comment]
Hi there.

Unfortunately, I have not gone anywhere on this project, not because of technical difficulties,

But because of having found out I have a heart condition, and trying to fight to get a neurological

Disorder resolved. So I’m really sorry I can’t give you more positive news. It left me off work for

Nearly 2 months so I’m catching up financially.

 

I’m just restocking certain electronic components this month so I may be able to restart the project in the next

Month or two.

 

I’ll keep you in touch though

 

Jay.
